app/domain/authentication/authn_azure/authenticator.rb

```diff
@@ -6,35 +6,77 @@ module AuthnAzure
     Err = Errors::Authentication::AuthnAzure
     Log = LogMessages::Authentication::AuthnAzure
     # Possible Errors Raised:
-    # TODO: Add errors
+    # TokenFieldNotFoundOrEmpty, MissingRequestParam
 
     Authenticator = CommandClass.new(
       dependencies: {
-        fetch_authenticator_secrets: Authentication::Util::FetchAuthenticatorSecrets.new,
-        logger:                      Rails.logger
+        fetch_authenticator_secrets:   Authentication::Util::FetchAuthenticatorSecrets.new,
+        verify_and_decode_token:       Authentication::AuthnOidc::VerifyAndDecodeToken.new,
+        logger:                        Rails.logger
       },
-      inputs: [:authenticator_input]
+      inputs:       [:authenticator_input]
     ) do
 
       extend Forwardable
-      def_delegators :@authenticator_input, :service_id, :authenticator_name, :account, :username, :request, :credentials
+      def_delegators :@authenticator_input, :service_id, :authenticator_name, :account, :credentials
+
+      JWT_FIELD_NAME = "jwt"
 
       def call
+        validate_credentials_include_azure_token
+        validate_azure_token
+        validate_required_token_fields_exist
         true
       end
 
       private
 
-      #def validate_azure_token
-      #
-      #end
-      #
-      #def validate_application_identity
-      #
-      #end
+      def validate_credentials_include_azure_token
+        # check that id token field exists and has some value
+        raise Errors::Authentication::RequestBody::MissingRequestParam, JWT_FIELD_NAME unless decoded_credentials.include?(JWT_FIELD_NAME) &&
+          !decoded_credentials[JWT_FIELD_NAME].empty?
+      end
+
+      def validate_azure_token
+        decoded_token
+      end
+
+      def decoded_token
+        @decoded_token ||= @verify_and_decode_token.call(
+          provider_uri:     provider_uri,
+          token_jwt:        decoded_credentials[JWT_FIELD_NAME],
+          claims_to_verify: {
+            verify_iss: true,
+            iss:        provider_uri
+          }
+        )
+      end
+
+      def validate_required_token_fields_exist
+        validate_token_field_exists( "xms_mirid")
+        validate_token_field_exists("oid")
+      end
+
+      def validate_token_field_exists field_name
+        @logger.debug(Log::ValidatingTokenFieldExists.new(field_name))
+        raise Err::TokenFieldNotFoundOrEmpty, field_name if decoded_token[field_name].to_s.empty?
+      end
+
+      def xms_mirid
+        @logger.debug(Log::ExtractedXmsMiridFromAzureToked.new(decoded_token["xms_mirid"]))
+        decoded_token["xms_mirid"]
+      end
+
+      def oid
+        decoded_token["oid"]
```

Comment:
why are we not able logging extracting oid? I don't believe this is sensitive info

Reply:
even if we don't want to print it out, I still think it is important for logging purposes 
`@logger.debug(Log::ExtractedXmsMiridFromAzureToked.new("oid"))`